x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/analytics/rate.yml

```diff
@@ -205,3 +205,125 @@
+          - '{"index": {}}'
+          - '{"@timestamp": "2021-04-28T18:50:05.000Z", "host": "one", "bytes_counter": 1350, "bytes_gauge": 40, "bytes": 40 }'
+          - '{"index": {}}'
+          - '{"@timestamp": "2021-04-28T18:50:06.000Z", "host": "one", "bytes_counter": 1420, "bytes_gauge": 70, "bytes": 70 }'
+          - '{"index": {}}'
+          - '{"@timestamp": "2021-04-28T18:50:07.000Z", "host": "one", "bytes_counter": 1500, "bytes_gauge": 80, "bytes": 80 }'
+          - '{"index": {}}'
+          - '{"@timestamp": "2021-04-28T18:50:08.000Z", "host": "one", "bytes_counter": 1520, "bytes_gauge": 20, "bytes": 20 }'
+          - '{"index": {}}'
+          - '{"@timestamp": "2021-04-28T18:50:09.000Z", "host": "one", "bytes_counter": 1550, "bytes_gauge": 30, "bytes": 30 }'
+
+  - do:
+      search:
+        index: test-rate
+        body:
+          size: 0
+          query:
+            bool:
+              filter:
+                range:
+                  "@timestamp":
+                    gte: "2021-04-28T18:00:00.000Z"
+                    lte: "2021-04-28T19:00:00.000Z"
+          aggs:
+            date_histogram:
+              date_histogram:
+                field: "@timestamp"
+                fixed_interval: 1m
+                time_zone: Europe/Ljubljana
+                min_doc_count: 1
+              aggs:
+                8:
+                  sum_bucket:
+                    buckets_path: 8-bucket>8-metric
+                9:
+                  sum_bucket:
+                    buckets_path: 9-bucket>9-metric
+                10:
+                  rate:
+                    field: bytes_gauge
+                    unit: second
+                11:
+                  rate:
+                    field: bytes
+                    unit: second
+                8-bucket:
+                  terms:
+                    field: host
+                    order:
+                      _key: desc
+                    size: 500
+                  aggs:
+                    8-metric:
+                      rate:
+                        field: bytes_gauge
+                        unit: second
+                9-bucket:
+                  time_series: { }
+                  aggs:
+                    9-metric:
+                      rate:
+                        field: bytes_counter
+                        unit: second
+
+  - match: { hits.total.value: 10 }
+  - length: { aggregations.date_histogram.buckets: 1 }
+  - match: { aggregations.date_histogram.buckets.0.key_as_string: "2021-04-28T20:50:00.000+02:00" }
+  - match: { aggregations.date_histogram.buckets.0.doc_count: 10 }
+  - close_to: { aggregations.date_histogram.buckets.0.11.value: { value: 25.83, error: 0.01 } }
+  - close_to: { aggregations.date_histogram.buckets.0.10.value: { value: 25.83, error: 0.01 } }
+  - close_to: { aggregations.date_histogram.buckets.0.9.value: { value: 0.06, error: 0.01 } }
```

Comment:
Might be worth inserting a comment here explaining why there's a difference? ie, because this agg is using the timestamps from the beginning and end of the bucket rather than the bucket width.